### PR TITLE
ScopeConfig for pydevlake

### PR DESCRIPTION
**Pull request by CamilleTeruel**
### Summary
Rename TransformationRule to ScopeConfig that also includes domain types.

### Does this close any open issues?
Part of #5159

### CODE REVIEW

**Comment by klesh** [comment]
@CamilleTeruel Seems like the test failed, can you fix them first? Thanks.

File: backend/test/helper/api.go [hunk 1]
```diff
@@ -194,43 +194,43 @@ func (d *DevlakeClient) GetScope(pluginName string, connectionId uint64, scopeId
...
 }
 
-func (d *DevlakeClient) GetTransformationRule(pluginName string, connectionId uint64, txRuleId uint64) any {
+func (d *DevlakeClient) GetScopeConfig(pluginName string, connectionId uint64, scopeConfigId uint64) any {
 	return sendHttpRequest[any](d.testCtx, d.timeout, debugInfo{
 		print:      true,
 		inlineJson: false,
-	}, http.MethodGet, fmt.Sprintf("%s/plugins/%s/connections/%d/transformation_rules/%d",
-		d.Endpoint, pluginName, connectionId, txRuleId), nil, nil)
+	}, http.MethodGet, fmt.Sprintf("%s/plugins/%s/connections/%d/scope_configs/%d",
```
Comment (keon94) [inline backend/test/helper/api.go:232]: I think this needs to be scope-config? I believe that's the convention other plugins are going to use

File: backend/test/helper/api.go [hunk 2]
```diff
@@ -194,8 +194,8 @@ func (d *DevlakeClient) GetScope(pluginName string, connectionId uint64, scopeId
 	}, http.MethodGet, fmt.Sprintf("%s/plugins/%s/connections/%d/scopes/%s?blueprints=%v", d.Endpoint, pluginName, connectionId, scopeId, listBlueprints), nil, nil)
 }
 
-func (d *DevlakeClient) DeleteScope(pluginName string, connectionId uint64, scopeId string, deleteDataOnly bool) []models.Blueprint {
-	return sendHttpRequest[[]models.Blueprint](d.testCtx, d.timeout, debugInfo{
+func (d *DevlakeClient) DeleteScope(pluginName string, connectionId uint64, scopeId string, deleteDataOnly bool) any {
```
Comment (keon94) [inline backend/test/helper/api.go:197]: You can just make this function return nothing (void)

**Comment by keon94** [comment]
There are some merge conflicts, otherwise LGTM.

**Review by keon94 — APPROVED** [review]

**Review by klesh — APPROVED** [review]
LGTM

**Pull request merged**